packages/file_selector/file_selector_windows/lib/src/dart_file_open_dialog_api.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ffi';
+
+import 'package:ffi/ffi.dart';
+import 'package:win32/win32.dart';
+
+/// FileOpenDialogAPI provider, it used to interact with an IFileOpenDialogInstance.
+class FileOpenDialogAPI {
```

Comment:
If this is intended to wrap `IFileOpenDialog`, why is it taking the dialog as an argument to every method instead of that being a constructor argument?